src/node_buffer.cc

```diff
@@ -602,6 +602,83 @@ void Compare(const FunctionCallbackInfo<Value> &args) {
 }
 
 
+void IndexOfString(const FunctionCallbackInfo<Value>& args) {
+  ASSERT(args[0]->IsObject());
+  ASSERT(args[1]->IsString());
+  ASSERT(args[2]->IsNumber());
+
+  ARGS_THIS(args[0].As<Object>());
+  node::Utf8Value str(args.GetIsolate(), args[1]);
+  uint32_t offset = args[2]->Uint32Value();
+
+  if (str.length() == 0) {
+    return args.GetReturnValue().Set(
+        MIN(static_cast<uint32_t>(obj_length), offset));
+  }
+  if (str.length() + offset > obj_length || obj_length == 0) {
+    return args.GetReturnValue().Set(-1);
+  }
+
+  char* result = strstr(obj_data + offset, *str);
+  args.GetReturnValue().Set(
+      result ? static_cast<int32_t>(result - obj_data) : -1);
```

Comment:
Maybe add a `if (result != nullptr) CHECK_LT(result - obj_data, INT_MAX);` one line up, in case the external array limit gets increased to something that doesn't fit in an int32_t.

(And maybe assign the result of `result - obj_data` to a variable, to avoid the repetition.)